log_outgoing_requests/handlers.py

```diff
@@ -2,43 +2,66 @@
 import traceback
 from urllib.parse import urlparse
 
-from django.conf import settings
+from .validators import validate_content
 
 
 class DatabaseOutgoingRequestsHandler(logging.Handler):
     def emit(self, record):
-        if settings.LOG_OUTGOING_REQUESTS_DB_SAVE:
+        from .models import OutgoingRequestsLogConfig
+
+        config = OutgoingRequestsLogConfig.get_solo()
+
+        if config.save_logs_enabled:
             from .models import OutgoingRequestsLog
 
-            trace = None
-
-            # save only the requests coming from the library requests
-            if record and record.getMessage() == "Outgoing request":
-                safe_req_headers = record.req.headers.copy()
-
-                if "Authorization" in safe_req_headers:
-                    safe_req_headers["Authorization"] = "***hidden***"
-
-                if record.exc_info:
-                    trace = traceback.format_exc()
-
-                parsed_url = urlparse(record.req.url)
-                kwargs = {
-                    "url": record.req.url,
-                    "hostname": parsed_url.hostname,
-                    "params": parsed_url.params,
-                    "status_code": record.res.status_code,
-                    "method": record.req.method,
-                    "req_content_type": record.req.headers.get("Content-Type", ""),
-                    "res_content_type": record.res.headers.get("Content-Type", ""),
-                    "timestamp": record.requested_at,
-                    "response_ms": int(record.res.elapsed.total_seconds() * 1000),
-                    "req_headers": self.format_headers(safe_req_headers),
-                    "res_headers": self.format_headers(record.res.headers),
-                    "trace": trace,
-                }
-
-                OutgoingRequestsLog.objects.create(**kwargs)
+            trace = ""
+
+            # skip requests not coming from the library requests
+            if not record or not record.getMessage() == "Outgoing request":
+                return
+
+            scrubbed_req_headers = record.req.headers.copy()
+
+            if "Authorization" in scrubbed_req_headers:
+                scrubbed_req_headers["Authorization"] = "***hidden***"
+
+            if record.exc_info:
+                trace = traceback.format_exc()
+
+            parsed_url = urlparse(record.req.url)
+            kwargs = {
+                "url": record.req.url,
+                "hostname": parsed_url.hostname,
```

Comment:
I agree that `netloc` would be preferable since it includes the port number if present, otherwise it's equivalent to `hostname`. However, since the model already includes a field for `hostname`, does it make sense to add this on top? Or should `hostname` be replaced with `netloc`?

Reply:
Discussed: replace `hostname` with `netloc`